src/ws/cockpithandlers.c

```diff
@@ -341,6 +341,28 @@ build_environment (GHashTable *os_release)
   if (ca_path)
     json_object_set_string_member (object, "CACertUrl", "/ca.cer");
 
+  gchar *contents = NULL;
+  gsize len;
+  GError *error = NULL;
+
+  const gchar *banner = cockpit_conf_string ("Session", "Banner");
+  if (banner)
+    {
+      // TODO: parse macros (see `man agetty` for possible macros)
+      g_file_get_contents (banner, &contents, &len, &error);
+    }
+
+  if (error)
```

Comment:
Please move that entire block into the `if`. Otherwise it's being run also if Banner is disabled. In particular, this sets the `json_object_set_string_member` to a NULL value. This shouldn't harm, but it's a bit confusing.